src/app/core/services/data-integration/videos.service.ts

```diff
@@ -0,0 +1,207 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, forkJoin, Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import _ from 'lodash';
+
+import { ShownVideo } from '../../../shared/interfaces/shown-video.interface';
+import { YtDataService } from './yt-data.service';
+import { VimeoDataService } from './vimeo-data.service';
+import { SavedVideoData } from '../../../shared/interfaces/saved-video-data.interface';
+import { VideosStoreService } from './videos-store.service';
+import { YT_VIDEO_TYPE } from '../../../shared/constans/videos-types';
+import { SORT_DESCENDING } from '../../../shared/constans/sort-values';
+import { VideoNotSaved } from '../../../shared/interfaces/video-not-saved.interface';
+import { SnackbarService } from '../utils/snackbar.service';
+import { DEMO_VIDEOS_ADDED_MESSAGE, VIDEO_NOT_FOUND_MESSAGE } from '../../../shared/constans/snackbar-messages';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class VideosService {
+
+  private videos: ShownVideo[] = [];
+  public sortType = SORT_DESCENDING;
+  public showOnlyFavourites = false;
+  public videos$ = new BehaviorSubject<ShownVideo[]>([]);
+  public isLoading$ = new BehaviorSubject<boolean>(false);
+
+  constructor(
+    private ytDataService: YtDataService,
+    private vimeoDataService: VimeoDataService,
+    private videosStoreService: VideosStoreService,
+    private snackbarService: SnackbarService,
+  ) {}
+
+  private getVideos(savedVideosData: SavedVideoData[]): Observable<ShownVideo[]> {
+    const ytIds = [];
+    const vimeoIds = [];
+    let foundVideos: ShownVideo[] = [];
+
+    this.isLoading$.next(true);
+
+    savedVideosData.forEach((video: SavedVideoData) => {
+      if (video.type === YT_VIDEO_TYPE) {
+        ytIds.push(video.id);
+      } else {
+        vimeoIds.push(video.id);
+      }
+    });
+
+    const ytObservable = this.ytDataService.getVideoByIds(ytIds);
+    const vimeoObservable = this.vimeoDataService.getVideoByIds(vimeoIds);
+
+    return forkJoin([ytObservable, vimeoObservable])
+      .pipe(
+        map((videos: any) => {
+          const [ytVideos, vimeoVideos] = videos;
+
+          foundVideos = foundVideos.concat(ytVideos, vimeoVideos);
+          foundVideos = foundVideos.map((video: ShownVideo) => {
+            const videoDataFromLocalStorage = savedVideosData.find(videoData => videoData.id === video.id);
+
+            video.addedToLibraryAt = videoDataFromLocalStorage.addedToLibraryAt;
+            video.isFavourite = videoDataFromLocalStorage.isFavourite;
+
+            return video;
+          });
+
+          return foundVideos;
+        })
+      );
+  }
+
+  public findVideoById(id: string): void {
+    const ytObservable = this.ytDataService.getVideoByIds([id]);
```

Comment:
Problem ogólny jest taki, że jak będziesz chciał dodać kolejny serwis do apki, jak Dailymotion, to będziesz musiał w wielu miejscach swojej apki dodawać dużo kodu. Czyli nie spełniłeś jednego z założeń apki. Ale już nie będziemy tego zmieniać bo pół apki będziesz musiał przemielić.